internal/clients/http/ec2/ec2_client.go

```diff
@@ -316,6 +317,29 @@ func (c *ec2Client) ListInstanceTypes(ctx context.Context) ([]*clients.InstanceT
 	return instances, nil
 }
 
+func (c *ec2Client) ListInstancesDescription(ctx context.Context, InstanceIds []string) ([]*clients.InstanceDescription, error) {
+	ctx, span := otel.Tracer(TraceName).Start(ctx, "ListInstancesDescription")
+	defer span.End()
+
+	input := &ec2.DescribeInstancesInput{
+		InstanceIds: InstanceIds,
+	}
+	resp, err := c.ec2.DescribeInstances(ctx, input)
+	if err != nil {
+		if isAWSUnauthorizedError(err) {
+			err = clients.UnauthorizedErr
+		}
+		span.SetStatus(codes.Error, err.Error())
+		return nil, fmt.Errorf("cannot fetch instances description: %w", err)
+	}
+	instancesList, err := c.parseDescribeInstances(resp)
+	if err != nil {
+		span.SetStatus(codes.Error, err.Error())
+		return nil, fmt.Errorf("failed to fetch instances description: %w", err)
+	}
+	return instancesList, nil
+}
```

Comment:
I would probably keep the naming convention from the AWS API here, even if it returns a slice. The call is named "DescribeInstances" so we could have the same name in our own interface too. Something like "DescribeInstanceDetails" maybe? (I like "Details" more than "Description" so the struct type could be `InstanceDetail`)

Reply:
Another idea for the name as I commented down below: `ReservationInstance`